File: 프로그래머스/1/12931. 자릿수 더하기/자릿수 더하기.py
def solution(n):
    answer = 0
    
    l = len(str(n))
    
#     한글자 일때 바로 반환
    if l == 1:
        return n
#     두 글자 이상일 땐 나누는 수에 10 씩 곱해가며 자리수 별 숫자 찾기
    for i in range(1, l+1):
        num = 10**i
        tmp = int((n % num)/10**(i-1))
        answer += tmp
    return answer

File: 프로그래머스/1/12931. 자릿수 더하기/자릿수 더하기.cs
using System;

public class Solution {
    public int solution(int n) {
        int answer = 0;
        
        int l = n.ToString().Length;
        
        if(l==1)
        {
            return n;
        }
        
        for (int i = 1; i<=l; i++)
        {
            int div = (int)Math.Pow(10,i);
            int tmp = (n % div) / (int)Math.Pow(10, i-1);
            answer += tmp;
        }
        
        return answer;
    }
}